plone/app/event/portlets/portlet_events.py

```diff
@@ -182,7 +182,11 @@ def events(self):
                 ret_mode=RET_MODE_ACCESSORS,
                 expand=True, limit=data.count, **query
             )
-        return events
+            eventlist=[]
+            for ev in events:
+                hasimage = True if getattr(ev.context, 'image', None) else False
```

Comment:
btw, this changes it only for the second (else) part if the ICollection condition above. I assume this wont work with collections.